### PR TITLE
nit: Refactor enterprisePackage target

### PR DESCRIPTION
**Pull request by Obero**
Warn: I'm targeting your `wip` branch

### CODE REVIEW

File: src/main/java/io/gatling/mojo/EnterprisePackageMojo.java [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/EnterprisePackageMojo.java:1]: This line is going to be restored by spotless.

File: src/main/java/io/gatling/mojo/EnterprisePackageMojo.java [hunk 2]
```diff
@@ -137,6 +140,31 @@ private Set<Artifact> resolveTransitively(Artifact artifact) {
...
+        MojoUtils.findByGroupId(allDeps, GATLING_HIGHCHARTS_GROUP_ID);
+    List<Artifact> allGatlingDeps = new ArrayList<>();
+
+    allGatlingDeps.addAll(depsWithGatlingGroupId);
+    allGatlingDeps.addAll(depsWithGatlingHighchartsGroupId);
+
+    return allGatlingDeps;
+  }
+
+  private static Artifact getGatlingApp(List<Artifact> depsWithGatlingGroupId) {
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/EnterprisePackageMojo.java:147]: We don't really care about locating the gatling-app jar. We care about getting the Gatling version.

File: src/main/java/io/gatling/mojo/EnterprisePackageMojo.java [hunk 3]
```diff
@@ -38,47 +37,17 @@
...
-            .filter(artifact -> MojoUtils.artifactNotIn(artifact, gatlingAndTransitiveDependencies))
-            .map(
-                artifact ->
-                    new Dependency(
-                        artifact.getGroupId(),
-                        artifact.getArtifactId(),
-                        artifact.getVersion(),
-                        artifact.getFile()))
-            .collect(Collectors.toSet());
+    String gatlingVersion = getGatlingApp(depsWithGatlingGroupId).getVersion();
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/EnterprisePackageMojo.java:51]: See comment below, we should have a getGatlingVersion instead.

File: src/main/java/io/gatling/mojo/EnterprisePackageMojo.java [hunk 4]
```diff
@@ -137,6 +140,31 @@ private Set<Artifact> resolveTransitively(Artifact artifact) {
...
+  }
+
+  private static Artifact getGatlingApp(List<Artifact> depsWithGatlingGroupId) {
+    Artifact gatlingApp =
+        depsWithGatlingGroupId.stream()
+            .filter(artifact -> artifact.getArtifactId().equals("gatling-app"))
+            .findFirst()
+            .orElse(null);
+
+    Objects.requireNonNull(gatlingApp, "Can't find Gatling libraries in dependencies");
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/EnterprisePackageMojo.java:154]: Note: requireNonNull returns the checked parameter itself.

**Review by slandelle — CHANGES_REQUESTED** [review]

File: src/main/java/io/gatling/mojo/EnterprisePackageMojo.java (hunk 4 above)
Comment (Obero) [reply src/main/java/io/gatling/mojo/EnterprisePackageMojo.java]: I simplified this part a bit further.
I just kept the `NullPointerException` in order to not change the current behaviour.

**Review by slandelle — APPROVED** [review]

**Pull request merged**